### PR TITLE
[SPARK-27880][SQL] Add  bool_and for every and bool_or for any as function aliases

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

bool_or(x) <=> any/some(x) <=> max(x)
bool_and(x) <=> every(x) <=> min(x)
Args:
  x: boolean
### Why are the changes needed?

PostgreSQL, Presto and Vertica, etc also support this feature:
### Does this PR introduce any user-facing change?

add new functions support

### How was this patch tested?

add ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @wangyum @cloud-fan @gatorsmile

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala [hunk 1]
```diff
@@ -316,6 +316,8 @@ object FunctionRegistry {
     expression[EveryAgg]("every"),
     expression[AnyAgg]("any"),
     expression[SomeAgg]("some"),
+    expression[BoolAnd]("bool_and"),
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala]: is `bool_and` just an alias of `every`? If so we can do `expression[EveryAgg]("bool_and")`

Comment (yaooqinn) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala]: Yes, it is. https://github.com/apache/spark/pull/26126/files#diff-fdccd9945d709da6c561b67a2e46a0d8R296, is there a way to keep the bool_and string in the schema?

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala]: We don't need to keep the function alias in the schema. There are many examples in `FunctionRegistry`, that a function has more than one name and we always use one name when displaying.

Comment (yaooqinn) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala]: I see. Then `any`, `some` and `bool_or` are mathematically equivalent, I guess we can only left `AnyAgg ` and remove other implemtations

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
Can we make the title more explicit that this just adds function alias?

**Comment by yaooqinn** [comment]
retest this please

**Review by wangyum — COMMENTED** [review]
We should enable these tests: [aggregates_part2.sql#L116-L160](https://github.com/apache/spark/blob/0a097426dc54ddb8b32b4802456c3e9b1ca7113a/sql/core/src/test/resources/sql-tests/inputs/postgreSQL/aggregates_part2.sql#L116-L160) and [udf-aggregates_part2.sql#L118-L162](https://github.com/apache/spark/blob/0a097426dc54ddb8b32b4802456c3e9b1ca7113a/sql/core/src/test/resources/sql-tests/inputs/udf/postgreSQL/udf-aggregates_part2.sql#L118-L162). Are we doing this in this PR or in the follow-up PR?

**Comment by yaooqinn** [comment]
@wangyum thanks for noticing. But why these same tests in two different test files?

**Pull request closed without merging**